File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode {
    if len(nums) == 0 {
        return nil
    }
    
    medianIndex := len(nums) / 2
    rootNode := &TreeNode{
        Val: nums[medianIndex],
    }
    left := sortedArrayToBST(nums[:medianIndex])
    right := sortedArrayToBST(nums[medianIndex+1:])
    
    rootNode.Left = left
    rootNode.Right = right
    
    return rootNode
}

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        if len(nums) == 0:
            return None
        
        median_index = len(nums)//2
        node = TreeNode(
            val=nums[median_index]
        )
        
        node.left = self.sortedArrayToBST(nums[:median_index])
        node.right = self.sortedArrayToBST(nums[median_index+1:])
        
        return node
            